### DISCUSSION TITLE
what are the handlers that are currently supported by the lsp server?

### DISCUSSION THREAD

**Discussion by Wordluc** [dotnet/roslyn#76518] [Q&A] [answered]
I've been trying to call [DidChangeWatchedFiles Notification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeWatchedFiles) in neovim, but i get the error: not found: "workspace/didChangeWatchedFiles" request handler for client "roslyn"
#76202 
.

**Comment by CyrusNajmabadi** [comment, marked as answer]
We don't have a fixed list. You'd have to examine the source code to find out. 

Our server also reports its capabilities, for parts of the spec that are optional so that the client can know.  

Here's the code for our handlers: https://github.com/dotnet/roslyn/tree/main/src/LanguageServer/Protocol/Handler
